File: kts_provider/src/main/java/com/lasley/kts_provider/helpers/UriParser.java
package com.lasley.kts_provider.helpers;


import android.net.Uri;

import java.util.HashMap;
import java.util.Map;

class UriParser {
    public static Map<UriToken, Object> parse(Uri input) {
        HashMap<UriToken, Object> data = new HashMap<>();
        data.put(UriToken.Scheme, input.getScheme());
        data.put(UriToken.Authority, input.getAuthority());
        data.put(UriToken.Path, input.getPath());
        data.put(UriToken.PathSegments, input.getPathSegments());
        data.put(UriToken.LastSegment, input.getLastPathSegment());
        data.put(UriToken.Query, input.getQuery());

        return data;
    }
}


enum UriToken {
    Scheme, Authority,
    Path, PathSegments, LastSegment,
    Query,
}

File: kts_viewer/src/main/kotlin/com/lasley/kts_viewer/data/UriParser.kt
package com.lasley.kts_viewer.data

import android.net.Uri

val Uri.toMap: Map<UriToken, Any?>
    get() = buildMap {
        put(UriToken.Scheme, scheme)
        put(UriToken.Authority, authority)
        put(UriToken.Path, path)
        put(UriToken.PathSegments, pathSegments)
        put(UriToken.LastSegment, lastPathSegment)
        put(UriToken.Query, query)
    }


enum class UriToken {
    Scheme, Authority,
    Path, PathSegments, LastSegment,
    Query,
}
